data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/AbstractNetworkElementRangeAction.java

```diff
@@ -21,15 +26,46 @@
  *
  * @author Viktor Terrier {@literal <viktor.terrier at rte-france.com>}
  */
-public abstract class AbstractNetworkElementRangeAction implements ApplicableRangeAction {
+public abstract class AbstractNetworkElementRangeAction extends AbstractRemedialAction implements RangeAction {
+
+    protected List<AbstractRange> ranges;
 
     protected NetworkElement networkElement;
 
     @JsonCreator
-    public AbstractNetworkElementRangeAction(@JsonProperty("networkElement") NetworkElement networkElement) {
+    public AbstractNetworkElementRangeAction(@JsonProperty("id") String id,
+                                             @JsonProperty("name") String name,
+                                             @JsonProperty("operator") String operator,
+                                             @JsonProperty("usageRules") List<UsageRule> usageRules,
+                                             @JsonProperty("ranges") List<AbstractRange> ranges,
+                                             @JsonProperty("networkElement") NetworkElement networkElement) {
+        super(id, name, operator, usageRules);
+        this.ranges = ranges;
```

Comment:
we need to add in ranges an absolute range corresponding to the physical limitations of the network element

Reply:
this is now done with the "synchronize" function

Reply:
In fact it is not very clean as it is not possible to delete properly the range if we want to desynchronize the crac from the network. that's why rangeaction doesn't implement the Synchronizable interface